tpchgen/src/text.rs

```diff
@@ -80,14 +80,14 @@ impl TextPool {
         let syntax = distributions.grammar().random_value(random);
         let max_length = syntax.len();
 
-        for i in (0..max_length).step_by(2) {
-            match syntax.chars().nth(i).unwrap() {
+        for c in syntax.chars().take(max_length).step_by(2) {
+            match c {
                 'V' => Self::generate_verb_phrase(distributions, builder, random),
                 'N' => Self::generate_noun_phrase(distributions, builder, random),
                 'P' => {
                     let preposition = distributions.prepositions().random_value(random);
                     builder.append(preposition);
-                    builder.append(" the ");
+                    builder.append_bytes(b" the ");
```

Comment:
Another thing I did that might help was to distinguish appending constant slices (where the length is known at compile time) and actual variable length strings to let the compiler generate specialized code when possible